File: chrome/browser/history_clusters/java/src/org/chromium/chrome/browser/history_clusters/HistoryClustersTabHelper.java
// Copyright 2021 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.chrome.browser.history_clusters;

import androidx.annotation.Nullable;

import org.jni_zero.NativeMethods;

import org.chromium.content_public.browser.WebContents;

/** Bridge between the C++ and Java Chrome Memories HistoryClustersTabHelper. */
public class HistoryClustersTabHelper {
    /**
     * Notify Memories service that the current tab's URL has been copied.
     * @param webContents WebContents object associated with the current tab.
     */
    public static void onCurrentTabUrlCopied(@Nullable WebContents webContents) {
        // Note: WebContents requires native libraries to be loaded and initialized.
        if (webContents != null && !webContents.isDestroyed()) {
            HistoryClustersTabHelperJni.get().onCurrentTabUrlCopied(webContents);
        }
    }

    /**
     * Notify Memories service that the current tab's URL has been shared.
     * @param webContents WebContents object associated with the current tab.
     */
    public static void onCurrentTabUrlShared(@Nullable WebContents webContents) {
        // Note: WebContents requires native libraries to be loaded and initialized.
        if (webContents != null && !webContents.isDestroyed()) {
            HistoryClustersTabHelperJni.get().onCurrentTabUrlShared(webContents);
        }
    }

    @NativeMethods
    interface Natives {
        void onCurrentTabUrlCopied(WebContents contents);

        void onCurrentTabUrlShared(WebContents contents);
    }
}

File: chrome/browser/ui/views/side_panel/history_clusters/history_clusters_tab_helper.cc
// Copyright 2023 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "chrome/browser/ui/views/side_panel/history_clusters/history_clusters_tab_helper.h"

#include "chrome/browser/ui/views/side_panel/history_clusters/history_clusters_side_panel_controller_utils.h"
#include "content/public/browser/web_contents.h"
#include "content/public/browser/web_contents_observer.h"

namespace side_panel {

HistoryClustersTabHelper::HistoryClustersTabHelper(
    content::WebContents* web_contents)
    : content::WebContentsObserver(web_contents),
      content::WebContentsUserData<HistoryClustersTabHelper>(*web_contents),
      delegate_(side_panel::history_clusters::CreateDelegate(web_contents)) {}

HistoryClustersTabHelper::~HistoryClustersTabHelper() = default;

void HistoryClustersTabHelper::ShowJourneysSidePanel(const std::string& query) {
  DCHECK(delegate_);
  delegate_->ShowJourneysSidePanel(query);
}

WEB_CONTENTS_USER_DATA_KEY_IMPL(HistoryClustersTabHelper);

}  // namespace side_panel
